src/plugins/vis_augmenter/public/utils/utils.ts

```diff
@@ -13,10 +13,15 @@ import {
 } from '../../../../plugins/expressions/public';
 import { ISavedAugmentVis, SavedAugmentVisLoader, VisLayerFunctionDefinition } from '../';
 
-// TODO: provide a deeper eligibility check.
-// Tracked in https://github.com/opensearch-project/OpenSearch-Dashboards/issues/3268
-export const isEligibleForVisLayers = (vis: Vis): boolean => {
-  return vis.params.type === 'line';
+export const isEligibleForVisLayers = (vis: Vis, dimensions: VislibDimensions): boolean => {
+  const isDateHistogram =
+    vis.data.aggs?.byTypeName('date_histogram').length === 1 && vis.data.aggs?.aggs.length === 2;
```

Comment:
Can you explain what `vis.data.aggs?.aggs.length === 2` is doing? adding a comment in the code might be worthwhile too